### PR TITLE
Desktop: don't count completed to-dos in note counts when they are not shown

### PR DESCRIPTION
**Pull request by lightray22**
If completed to-dos are not shown, they should not be included in the note counts.  This makes to-do list functionality better as you can have the count show only the remaining to-dos.

### CODE REVIEW

File: CliClient/tests/models_Folder.js [hunk 1]
```diff
@@ -128,25 +128,42 @@ describe('models_Folder', function() {
 	}));
 
 	it('should add node counts', asyncTest(async () => {
```
Comment (laurent22) [inline CliClient/tests/models_Folder.js]: Let's not mix tests so please create a new test unit for this feature.

File: ReactNativeClient/lib/BaseApplication.js [hunk 2]
```diff
@@ -450,11 +450,16 @@ class BaseApplication {
 			refreshFolders = true;
 		}
 
-		if (this.hasGui() && ((action.type == 'SETTING_UPDATE_ONE' && action.key.indexOf('folders.sortOrder') === 0) || action.type == 'SETTING_UPDATE_ALL')) {
```
Comment (laurent22) [inline ReactNativeClient/lib/BaseApplication.js]: Why this change?

File: ReactNativeClient/lib/models/Folder.js [hunk 3]
```diff
@@ -107,15 +107,18 @@ class Folder extends BaseItem {
 
 	// Calculates note counts for all folders and adds the note_count attribute to each folder
 	// Note: this only calculates the overall number of nodes for this folder and all its descendants
-	static async addNoteCounts(folders) {
+	static async addNoteCounts(folders, showDone) {
```
Comment (laurent22) [inline ReactNativeClient/lib/models/Folder.js]: For consistency, please name the new parameter `includeCompletedTodos` with default value `true`

File: ReactNativeClient/lib/models/Folder.js [hunk 4]
```diff
@@ -107,15 +107,18 @@ class Folder extends BaseItem {
...
 			foldersById[f.id] = f;
+			f.note_count = 0;
 		});
 
 		const sql = `SELECT folders.id as folder_id, count(notes.parent_id) as note_count 
-			FROM folders LEFT JOIN notes ON notes.parent_id = folders.id 
-			GROUP BY folders.id`;
+			FROM folders LEFT JOIN notes ON notes.parent_id = folders.id`
+			+ (!showDone ? ' WHERE (notes.is_todo = 0 OR notes.todo_completed = 0)' : '')
+			+ ` GROUP BY folders.id`;
```
Comment (laurent22) [inline ReactNativeClient/lib/models/Folder.js]: Please move the result of this logic to a variable to avoid having to concenate the template (I guess we should add an eslint rule for this)

**Comment by laurent22** [comment]
Thanks for the pull request. Overall it looks good and other than the few comments I've left it should be good to merge.

Also please install the pre commit hook as there are several linter errors.

File: ReactNativeClient/lib/BaseApplication.js (hunk 2 above)
Comment (lightray22) [reply ReactNativeClient/lib/BaseApplication.js]: I did not see there was already a section for includeCompletedTodos, so I added one. I then notice that the check for action.type == 'SETTING_UPDATE_ALL' was duplicated in two sections so I combined into one.  I now notice that the original change was unneeded but also that the SETTING_UPDATE_ALL check is duplicated in quite a few other places as well.  In lieu of refactoring all of those I will just revert this file.

Comment (lightray22) [reply ReactNativeClient/lib/BaseApplication.js]: Looks like I was mistaken again, the instances above are for refreshNotes not refreshFolders.  Back to my original point - I made the change because the check for SETTING_UPDATE_ALL was duplicated twice (now 3 times) and it seemed better to separate it.

**Comment by lightray22** [comment]
I have the pre-commit lint running but it doesn't seem to find the same errors as TravisCI and I'm not sure why.

**Comment by laurent22** [comment]
All good, thank you @lightray22!
